core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala

```diff
@@ -448,7 +448,7 @@ trait JavaRDDLike[T, This <: JavaRDDLike[T, This]] extends Serializable {
    * combine step happens locally on the master, equivalent to running a single reduce task.
    */
   def countByValue(): java.util.Map[T, jl.Long] =
-    mapAsSerializableJavaMap(rdd.countByValue().map((x => (x._1, new jl.Long(x._2)))))
+    mapAsSerializableJavaMap(rdd.countByValue().mapValues(jl.Long.valueOf))
```

Comment:
See my reply at https://github.com/apache/spark/pull/10554#discussion_r48938992 -- it was because I then saw this was how `countByKey` had been implemented. I picked consistency, but it's as reasonable to change `countByKey` I suppose. Do you feel strongly about it one way or the other?